TestOtpCore/OtpAuthUriTest.cs

```diff
@@ -137,10 +219,49 @@ public void ToStringPreserve()
             Assert.Equal(6, uri.Digits);
             Assert.Equal(30, uri.Period);
             Assert.Equal(OtpHmacAlgorithm.HmacSha256, uri.Algorithm);
+            Assert.True(UriComparer.AreEqual(uri.ToString(), uriString));
+            // test 2 -- issuer and account may not contain colons
+            Assert.Throws<ArgumentException>(() =>
+                new OtpAuthUri(
+                    "otpauth://totp/Example:al:ice@google.com?Algorithm=sha256&Secret=JBSWY3DPEHPK3PXP"));
+            Assert.Throws<ArgumentException>(() =>
+                new OtpAuthUri(
+                    "otpauth://totp/Ex:ample%3Aalice@google.com?Algorithm=sha256&Secret=JBSWY3DPEHPK3PXP"));
+            Assert.Throws<ArgumentException>(() =>
+                new OtpAuthUri(
+                    "otpauth://totp/Example::alice@google.com?Algorithm=sha256&Secret=JBSWY3DPEHPK3PXP"));
+            // test 3 -- unless the issuer parameter also contains matching colons to remove ambiguity
+            uriString =
+                "otpauth://totp/ACME%3FCo%3D192.168.1.1%3A8080:%CE%B1ccount?secret=AE&issuer=ACME%3fCo%3d192.168.1.1%3a8080&algorithm=SHA1&period=30&digits=6";
+            uri = new OtpAuthUri(uriString);
+            Assert.NotNull(uri);
+            Assert.Equal(OtpType.Totp, uri.Type);
+            Assert.Equal("ACME?Co=192.168.1.1:8080", uri.Issuer);
+            Assert.Equal("αccount", uri.Account);
+            Assert.Equal("AE", uri.Secret);
+            Assert.Equal(6, uri.Digits);
+            Assert.Equal(30, uri.Period);
+            Assert.Equal(OtpHmacAlgorithm.HmacSha1, uri.Algorithm);
+            Assert.True(UriComparer.AreEqual(uri.ToString(), uriString));
+        }
+
+        [Fact]
+        public void GitHubIssueNo9Pt3()
+        {
+            Assert.Throws<ArgumentException>(() =>
+                new OtpAuthUri((OtpType)42, Encoding.ASCII.GetBytes("12345678901234567890"), "bob@example.corp"));
+        }
+
+        [Fact]
+        public void GitHubIssueNo9Pt4()
```

Comment:
@Kevin-Andrew
Here is the test for the case you specified that wasn't working with Google Authenticator.